### ISSUE TITLE
Training error after modifying nn.View and nn.Linear

### ISSUE THREAD

**Issue by juntawu** [XingangPan/SCNN#44]
I simply remove the nn.View(4500)  and insert a new nn.View(4500) in **SCNN-gen.lua**, as the following shows:

`       classifier = model:get(44):get(2)
        classifier:remove(3)
        classifier:insert(nn.View(4500), 3)
        classifier:remove(4)
        classifier:insert(nn.Linear(4500, 128), 4)
`
Error comes durining training:

`/home/lishanliao/torch/install/bin/luajit: ...shanliao/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 2 callback] .../lishanliao/torch/install/share/lua/5.1/nn/Container.lua:67: 
In 44 module of nn.Sequential:
In 2 module of nn.ConcatTable:
In 7 module of nn.Sequential:
/home/lishanliao/torch/install/share/lua/5.1/nn/THNN.lua:110: output and gradOutput have different number of elements: output[4] has 4 elements, while gradOutput[3] has 3 elements at /home/lishanliao/torch/extra/cunn/lib/THCUNN/generic/Sigmoid.cu:23
`
What's wrong???

**Comment by juntawu** [comment]
@XingangPan Can you release the code to generate vgg.t7 please ?

**Comment by virajdeokar** [comment]
Try doing 
`classifier: insert(nn.Linear(4500, 128), 3)`
instead

**Comment by cardwing** [comment]
@juntawu , @virajdeokar , please refer to this [143](https://github.com/facebook/fb.resnet.torch/issues/143)

**Comment by leofanpico** [comment]
@juntawu     What's your GPU's number and it's memory?

**Comment by juntawu** [comment]
@leofanpico 8 GPUs, 8 G per GPU
